### PR TITLE
Add cri-containerd test document for developer.

### PR DESCRIPTION
**Pull request by Random-Liu**
@mikebrow This PR added document for all kinds of test for `cri-containerd`.

Signed-off-by: Lantao Liu <lantaol@google.com>

### CODE REVIEW

File: docs/testing.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+CRI-Containerd Testing Guide
+============================
+This document assumes you have already setup the development environment (go, git, `cri-contaienrd` repo etc.).
```
Comment (mikebrow) [inline docs/testing.md]: /s/contaienrd/containerd/

File: docs/testing.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+CRI-Containerd Testing Guide
+============================
+This document assumes you have already setup the development environment (go, git, `cri-contaienrd` repo etc.).  
+ 
+Before sending pull requests you should at least make sure your changes have passed code verification, unit test and CRI validation test.
```
Comment (mikebrow) [inline docs/testing.md]: Before sending pull requests you should at least make sure your changes have passed verification, unit, and CRI validation tests.

File: docs/testing.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+## Code verification
```
Comment (mikebrow) [inline docs/testing.md]: Verification Test

File: docs/testing.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+Code verification enforces some styles on go code, boilterplate etc.
```
Comment (mikebrow) [inline docs/testing.md]: The verification test includes lint, code formatting, and boilerplate tests.

File: docs/testing.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+* Install tools used by code verification:
```
Comment (mikebrow) [inline docs/testing.md]: the verification test

File: docs/testing.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+```shell
+make install.tools
+```
+* Run code verification:
```
Comment (mikebrow) [inline docs/testing.md]: the verification test:

File: docs/testing.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+```shell
+make verify
+```
+## Unit test
+Run all unit tests in `cri-containerd` repo.
+```
+make test
+```
+## CRI validation test
```
Comment (mikebrow) [inline docs/testing.md]: CRI Validation Test

File: docs/testing.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+* Install tools used by code verification:
+```shell
+make install.tools
+```
+* Run code verification:
+```shell
+make verify
+```
+## Unit test
```
Comment (mikebrow) [inline docs/testing.md]: Unit Test

File: docs/testing.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+Run all unit tests in `cri-containerd` repo.
+```
+make test
+```
+## CRI validation test
+[CRI validation test](https://github.com/kubernetes/community/blob/master/contributors/devel/cri-validation.md) is a test framework validating that a Container Runtime Interface (CRI) implementation meets all the requirements.
```
Comment (mikebrow) [inline docs/testing.md]: /s/is a test framework ... /is an integration test that verifies whether `cri-containerd` uses `containerd` to correctly implement the Container Runtime Interface (CRI)./

File: docs/testing.md (hunk 3 above)
Comment (Random-Liu) [inline docs/testing.md]: It's not quite clear what we are verifying if it's only `Verification test`.

Comment (mikebrow) [inline docs/testing.md]: thus the explanation below.. don't need a description in the title.

File: docs/testing.md (hunk 1 above)
Comment (Random-Liu) [inline docs/testing.md]: Will do.

File: docs/testing.md (hunk 2 above)
Comment (Random-Liu) [inline docs/testing.md]: Will do.

File: docs/testing.md (hunk 3 above)
Comment (Random-Liu) [inline docs/testing.md]: We call it [`source code verification`](https://github.com/kubernetes-incubator/cri-containerd/blob/master/Makefile#L40) in Makefile, too.

If we generate a TOC in the future, I think `Code Verification` will be better than `Verification Test`. :)

File: docs/testing.md (hunk 4 above)
Comment (Random-Liu) [inline docs/testing.md]: Will do.

File: docs/testing.md (hunk 8 above)
Comment (Random-Liu) [inline docs/testing.md]: Will do.

File: docs/testing.md (hunk 7 above)
Comment (Random-Liu) [inline docs/testing.md]: Will do.

File: docs/testing.md (hunk 9 above)
Comment (Random-Liu) [inline docs/testing.md]: CRI validation test is not just for `cri-containerd`, and let's use similar definition with their document. :)

File: docs/testing.md (hunk 1 above)
Comment (Random-Liu) [inline docs/testing.md]: Done.

File: docs/testing.md (hunk 2 above)
Comment (Random-Liu) [inline docs/testing.md]: Done.

File: docs/testing.md (hunk 4 above)
Comment (Random-Liu) [inline docs/testing.md]: Done.

File: docs/testing.md (hunk 8 above)
Comment (Random-Liu) [inline docs/testing.md]: Done.

File: docs/testing.md (hunk 3 above)
Comment (mikebrow) [inline docs/testing.md]: That's why we call em opinions, everyone has one :-) To me it's the verification test for all files code, config, documentation etc.   lint, format, headers, etc for code, yaml, md files, yada yada

File: docs/testing.md (hunk 9 above)
Comment (mikebrow) [inline docs/testing.md]: yeah so is CRI Validation Test "make test-cri" for cri-containerd or is it cri-tools's `critest validation` as configured to verify a particular CRI implementation? I presumed you meant `make test-cri` for cri-containerd.

Comment (mikebrow) [inline docs/testing.md]: IOW we talk about unit test above wrt to cri-containerd.. then switch to CRI Validation test with respect to all CRI implementors.. Perspective moved.  It's easier to follow if the perspective / scope stays the same.

File: docs/testing.md (hunk 3 above)
Comment (Random-Liu) [inline docs/testing.md]: All of them are our code, in fact. :p

Comment (mikebrow) [inline docs/testing.md]: yeah you could argue they are all code but the term used is "verify" cause we like to use shorter terms when possible, no? :-)

File: docs/testing.md (hunk 9 above)
Comment (mikebrow) [inline docs/testing.md]: How about we call it "Integration Test"

**Pull request merged**